.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,138 @@
+name: Build and Test LLVM/MLIR
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    name: Build and Test
+    runs-on: self-hosted
+
+    steps:
+      # Clone the repo. Do shallow clone up until depth two to save clone time.
+      - name: Checkout Pluto
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - name: Checkout submodules
+        # Checkout submodule via the command so that we avoid a complete
+        # recursive checkout. We only need the submodules Pluto depends on.
+        run: git submodule update --init --depth 1
+
+      - name: Generate build
+        # Use a bash shell so we can use the same syntax for environment variable
+        # access regardless of the host operating system
+        shell: bash
+        run: ./autogen.sh
+
+      - name: Configure build
+        # Use a bash shell so we can use the same syntax for environment variable
+        # access regardless of the host operating system
+        shell: bash
+        run: ./configure
```

Comment:
``--enable-glpk`` to enable tests for GLPK and DFP suite? I am not sure if the server for CI has glpk-dev installed.

Reply:
We can install that if it's not there - I'll add it. The runner is running on mcastle6.